remill/Arch/AArch64/Arch.cpp

```diff
@@ -487,13 +487,19 @@ static void AddExtendRegOperand(Instruction &inst, RegClass rclass,
 
 static void AddImmOperand(Instruction &inst, uint64_t val,
                           ImmType signedness=kUnsigned,
-                          unsigned size=64) {
+                          unsigned size=64, bool is_float=false) {
   Operand op;
-  op.type = Operand::kTypeImmediate;
+  if (is_float) {
+    op.type = Operand::kTypeImmediateFloat;
+    op.imm_float.val = size == 64 ? static_cast<double_t>(val) : static_cast<float_t>(val);
```

Comment:
A static cast does the wrong thing here, what you want is a `reinterpret_cast`. I think it'd be simpler to spread things out into a `kTypeImmediateFloat32` and `kTypeImmediateFloat64`, and handle the casts separately.